### PR TITLE
refactor controller as posthook when configured and enabled

### PR DESCRIPTION
**Pull request by deads2k**
builds on https://github.com/kubernetes/kubernetes/pull/33785 .

Models the bootstrap controller as a PostHook which is added when its API group is available and its configured.

@liggitt Stefan is out for a while.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33841)

### CODE REVIEW

**Comment by deads2k** [comment]
@liggitt @ncdc its big, but all the prereqs have gone in.  Using a `poststarthook` to start this should help us re-use `master.go` downstream.

**Comment by ncdc** [comment]
I'll take a look tonight or tomorrow morning.

**Review by ncdc — CHANGES_REQUESTED** [review]
Mostly LGTM. Just a couple of minor issues.

**Comment by deads2k** [comment]
comments addressed.

File: pkg/master/controller.go [hunk 1]
```diff
@@ -72,6 +74,42 @@ type Controller struct {
 	runner *async.Runner
 }
 
+// NewBootstrapController returns a controller for watching the core capabilities of the maste
```
Comment (ncdc) [inline pkg/master/controller.go]: missing r at eol

**Pull request merged**